test/suite/test_tiered14.py

```diff
@@ -0,0 +1,153 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, random, wiredtiger, wtscenario, wttest
+from wtdataset import TrackedSimpleDataSet, TrackedComplexDataSet
+
+# test_tiered14.py
+#    Test somewhat arbitrary combinations of flush_tier, checkpoint, restarts,
+#    data additions and updates
+class test_tiered14(wttest.WiredTigerTestCase):
+    uri = "table:test_tiered14-{}"   # format for subtests
+
+    auth_token = "test_token"
+    bucket = "mybucket"
+    cachedir = "mybucket-cache"
+    bucket_prefix = "pfx_"
+    extension_name = "local_store"
+
+    multiplier = [
+        ('S', dict(multiplier=1)),
+#        ('M', dict(multiplier=10)),
+#        ('L', dict(multiplier=100)),
+#        ('XL', dict(multiplier=1000)),
+    ]
+    keyfmt = [
+#        ('integer', dict(keyfmt='i')),
+        ('string', dict(keyfmt='S')),
+    ]
+    scenarios = wtscenario.make_scenarios(multiplier, keyfmt)
+
+    def conn_config(self):
+        if not os.path.exists(self.bucket):
+            os.mkdir(self.bucket)
+        return \
+          'tiered_storage=(auth_token=%s,' % self.auth_token + \
+          'bucket=%s,' % self.bucket + \
+          'bucket_prefix=%s,' % self.bucket_prefix + \
+          'cache_directory=%s,' % self.cachedir + \
+          'name=%s),tiered_manager=(wait=0)' % self.extension_name
+
+    # Load the local store extension.
+    def conn_extensions(self, extlist):
+        # Windows doesn't support dynamically loaded extension libraries.
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('storage_sources', self.extension_name)
+
+    def progress(self, s):
+        self.verbose(3, s)
+        self.pr(s)
+
+    # Run a set of operations, indicated by a string.
+    #  a = add some number of keys
+    #  u = update some number of keys
+    #  c = checkpoint
+    #  r = reopen
+    #  f = flush_tier
+    #  . = check to make sure all expected values are present
+    #
+    # We require a unique test number so we get can generate a different uri from
+    # previous runs.  A different approach is to drop the uri, but then we need to
+    # remove the bucket and cache, which is specific to the storage source extension.
+    def playback(self, testnum, ops):
+        uri = self.uri.format(testnum)
+        self.progress('Running ops: {} using uri {}'.format(ops, uri))
+        ds = TrackedSimpleDataSet(self, uri, self.multiplier,
+                                  config='key_format=' + self.keyfmt)
+        #ds = TrackedComplexDataSet(self, self.uri, self.multiplier,
```

Comment:
Is the commented out use of `TrackedComplexDataSet` old code that should be removed?  Future enhancement for the test that maybe needs a comment? Something else?

Reply:
I fixed this - I made separate scenarios for TrackedComplexDataset and TrackedSimpleDataSet, and commented out the complex one.  WT-8758 will attempt to enable all the scenarios.